x-pack/functionbeat/provider/aws/config.go

```diff
@@ -23,18 +24,25 @@ type Config struct {
 const maxMegabytes = 3008
 
 // DefaultLambdaConfig confguration for AWS lambda function.
-var DefaultLambdaConfig = &lambdaConfig{
-	MemorySize:  128 * 1024 * 1024,
-	Timeout:     time.Second * 3,
-	Concurrency: 5,
-}
+var (
+	DefaultLambdaConfig = &lambdaConfig{
+		MemorySize:  128 * 1024 * 1024,
+		Timeout:     time.Second * 3,
+		Concurrency: 5,
+	}
+
+	arnRolePattern = "arn:(aws[a-zA-Z-]*)?:iam::\\d{12}:role/?[a-zA-Z_0-9+=,.@\\-_/]+"
```

Comment:
We are using the arn reference in a few places in the code, we might want to create a custom type for it and have a single place to validate it.

As a note, I've decided not to add a validation for arn, since I was a bit worried that we would get it wrong. Can you add a link concerning the rules for an ARN.